### ISSUE TITLE
Unknown variable errors in atlas on terraform push

### ISSUE THREAD

**Issue by fromonesrc** [hashicorp/terraform#1480]
I've tried setting aws key vars set in `terraform.tfvars`, `variables.local.tf` but I get the following error after a `terraform push -name foo/bar` triggers a plan execution.

```
Plan execution has failed

The plan encountered an errored while executing 2 minutes ago. Plans do not modify real infrastructure, so this failure is safe. The error is shown below.

There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * module root: 4 error(s) occurred:

* resource 'aws_instance.nat' config: unknown variable referenced: aws_key_name
* provider config 'aws': unknown variable referenced: aws_access_key
* provider config 'aws': unknown variable referenced: aws_secret_key
* resource 'aws_instance.bastion' config: unknown variable referenced: aws_key_name
```

I've also tried `terraform push -name foo/bar -var "aws_secret_key=foo"` and get the same error.

I'm running Terraform v0.4.1.

**Comment by fromonesrc** [comment]
The vars are set on Atlas 
![](http://cl.ly/image/383R3Q183V1A/Image%202015-04-10%20at%202.46.37%20PM.png)

**Comment by mitchellh** [comment]
I clarified the error message, but you need to define the variables with `variable.

**Comment by fromonesrc** [comment]
@mitchellh Thanks but I don't think this is resolved or I'm misunderstanding something key about Terraform/Atlas variables. The variables were defined in a `variable` block but the ones that were missing in Atlas were the vars I defined in `variables.local.tf` (and had them .gitignored since they are sensitive). 

I found that by moving those sensitive vars to `variables.tf`, Atlas now picks them up. Is this a bug in Atlas? Terraform can handle `variables.local.tf` fine locally but fails with the same var config on Atlas.

According to [Terraform's variable docs](https://terraform.io/intro/getting-started/variables.html), I think this should have worked: 

> Note that the file can be named anything, since Terraform loads all files ending in .tf in a directory.
